src/aerovaldb/utils/copy.py

```diff
@@ -0,0 +1,43 @@
+from .. import AerovalDB, open, AccessType
+from ..utils import async_and_sync
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+@async_and_sync
+async def copy_db_contents(source: str | AerovalDB, dest: str | AerovalDB):
+    """
+    Utility function for copying the contents of one db connection to another
+    Currently this implementation requires the destination db to be empty.
+
+    :param source : Instance of AerovalDB or resource string passed to
+        aerovaldb.open()
+    :param dest : Instance of AerovalDB or resource string passed to
+        aerovaldb.open()
+
+    :raises : ValueError
+        If destination is not empty.
+    :raises : IOError
+        Number of items in destination is not the same as the source after copy.
+    """
+    if isinstance(source, str):
+        source = open(source)
+
+    if isinstance(dest, str):
+        dest = open(dest)
+
+    if len(await dest.list_all()) > 0:
+        ValueError("Destination database is not empty.")
+
+    for i, uri in enumerate(await source.list_all()):
+        logger.info(f"Processing item {i} of {len(await source.list_all())}")
+        data = await source.get_by_uri(uri, access_type=AccessType.JSON_STR)
+        await dest.put_by_uri(data, uri)
+
+    dst_len = len(await dest.list_all())
+    src_len = len(await source.list_all())
+    if dst_len != src_len:
+        raise IOError(
+            f"Unexpected number of items in destination after copy. Expected {src_len}, got {dst_len}"
+        )
```

Comment:
can you create a installed small command-line utility `aerovaldb_cp` from this function. Please use argparse or typer for arguments and help.

Reply:
Is this something you want me to do as part of this PR, or should I create an issue for it?